parlai/core/torch_classifier_agent.py

```diff
@@ -293,96 +304,13 @@ def eval_step(self, batch):
             loss = self.criterion(scores, labels)
             self.record_local_metric('loss', AverageMetric.many(loss))
             loss = loss.mean()
-            self._update_confusion_matrix(batch, preds)
+            self._update_confusion_matrix(preds, batch)
 
         if self.opt.get('print_scores', False):
             return Output(preds, probs=probs.cpu())
         else:
             return Output(preds)
 
-    def reset_metrics(self):
-        """
-        Reset metrics.
-        """
-        super().reset_metrics()
-        self.metrics['confusion_matrix'] = defaultdict(int)
-
-    def _report_prec_recall_metrics(self, confmat, class_name, metrics):
-        """
-        Use the confusion matrix to compute precision and recall.
-
-        :param confmat:
-            the confusion matrics
-        :param str class_name:
-            the class name to compute P/R for
-        :param metrics:
-            metrics dictionary to modify
-        :return:
-            the number of examples of each class.
-        """
-        # TODO: document these parameter types.
-        eps = 0.00001  # prevent divide by zero errors
```

Comment:
I probably missed this - we don't need this tolerance anymore because we're doing more robust divide-by-zero checking in the new `Metric`s themselves?